src/BuildScriptGenerator/Node/NodePlatform.cs

```diff
@@ -163,6 +182,36 @@ public PlatformDetectorResult Detect(RepositoryContext context)
                 packageInstallerVersionCommand = NodeConstants.NpmVersionCommand;
             }
 
+            // If a 'lerna.json' file exists, override the npm client that lerna chosen to build monorepo.
+            if (nodePlatformDetectorResult.HasLernaJsonFile)
+            {
+                packageManagerCmd = nodePlatformDetectorResult.LernaNpmClient;
+                runBuildLernaCommand = string.Format(
+                        NodeConstants.PkgMgrRunBuildCommandTemplate,
+                        NodeConstants.LernaCommand);
+                if (nodePlatformDetectorResult.LernaNpmClient.Equals(
+                    NodeConstants.YarnCommand, StringComparison.OrdinalIgnoreCase))
+                {
+                    packageInstallCommand = NodeConstants.YarnPackageInstallCommand;
+                    configureYarnCache = true;
+                    packageInstallerVersionCommand = NodeConstants.YarnVersionCommand;
+                    installLernaCommand = NodeConstants.InstallLernaCommandYarn;
+                }
+                else
+                {
+                    packageInstallCommand = NodeConstants.NpmPackageInstallCommand;
+                    packageInstallerVersionCommand = NodeConstants.NpmVersionCommand;
+                    installLernaCommand = NodeConstants.InstallLernaCommandNpm;
+                }
+            }
+
+            // If a 'lage.config.js' file exits, run build using lage specifc commands.
+            if (nodePlatformDetectorResult.HasLageConfigJSFile)
+            {
+                runBuildLageCommand = ctx.SourceRepo.FileExists(NodeConstants.YarnLockFileName) ?
+                        NodeConstants.YarnRunLageBuildCommand : NodeConstants.NpmRunLageBuildCommand;
+            }
+
```

Comment:
line 185 to line 214  (except 171 to 183) can be moved up as else if condition at line 170

Reply:
I put them at here because we want to use lerna chosen npm client not from what detect. for example, if `yarn.lock` found but lerna have `npmClient: npm` line in `lerna.json` we still choose npm. It will be after the file check so it's able to override.